river/ensemble/adaptive_random_forest.py

```diff
@@ -0,0 +1,824 @@
+import abc
+import collections
+import math
+import typing
+import copy
+
+import numpy as np
+
+from river import base
+from river.drift import ADWIN
+from river.metrics import Accuracy
+from river.metrics import MSE
+from river.metrics import MAE
+from river.metrics.base import MultiClassMetric
+from river.metrics.base import RegressionMetric
+from river.tree.arf_hoeffding_tree_classifier import ARFHoeffdingTreeClassifier
+from river.tree.arf_hoeffding_tree_regressor import ARFHoeffdingTreeRegressor
+from river.utils.skmultiflow_utils import check_random_state
+
+
+class BaseForest(base.EnsembleMixin):
+
+    _FEATURES_SQRT = "sqrt"
+    _FEATURES_LOG2 = "log2"
+
+    def __init__(self,
+                 n_models: int,
+                 max_features: typing.Union[bool, str, int],
+                 lambda_value: int,
+                 drift_detector: typing.Union[base.DriftDetector, None],
+                 warning_detector: typing.Union[base.DriftDetector, None],
+                 metric: typing.Union[MultiClassMetric, RegressionMetric],
+                 disable_weighted_vote,
+                 seed):
+        super().__init__([None])  # List of models is properly initialized later
+        self.models = []
+        self.n_models = n_models
+        self.max_features = max_features
+        self.lambda_value = lambda_value
+        self.metric = metric
+        self.disable_weighted_vote = disable_weighted_vote
+        self.drift_detector = drift_detector
+        self.warning_detector = warning_detector
+        self.seed = seed
+        self._rng = check_random_state(self.seed)   # Actual random number generator
+
+        # Internal parameters
+        self._n_samples_seen = 0
+        self._base_member_class = None
+
+    def learn_one(self, x: dict, y: base.typing.ClfTarget, **kwargs):
```

Comment:
Should we include typing annotation for the classification case here too?

Reply:
Something like the generic `base.typing.Target`?

Reply:
Nice catch, I changed it to `base.typing.Target` in 53c75bb9b45bfa0539d0c4bcd86bd3707603c5d9